rocketpy/tools.py

```diff
@@ -549,6 +550,119 @@ def generate_monte_carlo_ellipses_coordinates(
     return outputs
 
 
+def load_monte_carlo_data(
+    input_filename,
+    output_filename,
+    parameters_list,
+    target_variables_list,
+):
+    """Reads MonteCarlo simulation data file and builds parameters and flight
+    variables matrices from specified
+
+    Parameters
+    ----------
+    input_filename : str
+        Input file exported by MonteCarlo class. Each line is a
+        sample unit described by a dictionary where keys are parameters names
+        and the values are the sampled parameters values.
+
+    output_filename : str
+        Output file exported by MonteCarlo.simulate function. Each line is a
+        sample unit described by a dictionary where keys are target variables
+        names and the values are the obtained values from the flight simulation.
+
+    parameters_list : list[str]
+        List of parameters whose values will be extracted.
+
+    target_variables_list : list[str]
+        List of target variables whose values will be extracted.
+
+    Returns
+    -------
+    parameters_matrix: np.matrix
+        Numpy matrix contaning input parameters values. Each column correspond
+        to a parameter in the same order specified by 'parameters_list' input.
+
+    target_variables_matrix: np.matrix
+        Numpy matrix contaning target variables values. Each column correspond
+        to a target variable in the same order specified by 'target_variables_list'
+        input.
+    """
+    number_of_samples_parameters = 0
+    number_of_samples_variables = 0
+
+    # Auxiliary function that unnests dictionary
+    def unnest_dict(x):
+        new_dict = {}
+        for key, value in x.items():
+            # the nested dictionary is inside a list
+            if isinstance(x[key], list):
+                # sometimes the object inside the list is another list
+                # we must skip these cases
+                if isinstance(value[0], dict):
+                    inner_dict = unnest_dict(value[0])
+                    inner_dict = {
+                        key + "_" + inner_key: inner_value
+                        for inner_key, inner_value in inner_dict.items()
+                    }
+                    new_dict.update(inner_dict)
+            else:
+                new_dict.update({key: value})
+
+        return new_dict
```

Comment:
I would define this function outside the `load_monte_carlo_data()` function, this would allow us to re-use this function in other contexts.

Also, the term `flatten` is usually used to describe this kid of "parse a nested dictionary" operation. Maybe it would be a good alternative of name.

Reply:
`flatten_dict` is definitely a better name for it! With respect to defining it outside the `load_monte_carlo_data`, there is a detail in the implementation that is specific to the way `MonteCarlo` simulation data is saved. I have not defined it outside of the function in this first review round because of this, but I might think of a way to make it general and work with the `MonteCarlo` data in the future.